src/hotspot/cpu/x86/x86.ad

```diff
@@ -8272,6 +8260,45 @@ instruct loadIotaIndices(vec dst, immI_0 src, rRegP scratch) %{
   ins_pipe( pipe_slow );
 %}
 
+#ifdef _LP64
+instruct VectorPopulateIndex(vec dst, rRegI src1, immI_1 src2, vec vtmp, rRegP scratch) %{
+  match(Set dst (PopulateIndex src1 src2));
+  effect(TEMP dst, TEMP vtmp, TEMP scratch);
+  format %{ "vector_populate_index $dst $src1 $src2\t! using $vtmp and $scratch as TEMP" %}
+  ins_encode %{
+     assert($src2$$constant == 1, "required");
+     int vlen = Matcher::vector_length(this);
```

Comment:
May I ask why use `Matcher::vector_length()` here, rather than `Matcher::vector_length_in_bytes()`, for `load_iota_indices()`? Thanks.